packages/contracts-periphery/contracts/foundry-tests/Optimist.t.sol

```diff
@@ -0,0 +1,371 @@
+            val: bytes(base_uri)
+        });
+        vm.prank(alice_admin);
+        attestationStation.attest(attestationData);
+
+        // mint an NFT
+        vm.prank(bob);
+        optimist.mint(bob);
+
+        // assert tokenURI is set
+        assertEq(optimist.baseURI(), base_uri);
+        assertEq(
+            optimist.tokenURI(256),
+            // solhint-disable-next-line max-line-length
+            "https://storageapi.fleek.co/6442819a1b05-bucket/optimist-nft/attributes/0x0000000000000000000000000000000000000100.json"
+        );
+    }
+
+    /**
+     * @dev Should return a boolean of if the address is whitelisted
+     */
+    function test_optimist_is_whitelisted() external {
+        // whitelist bob
+        AttestationStation.AttestationData[]
+            memory attestationData = new AttestationStation.AttestationData[](1);
+        // we are using true but it can be any non null value
+        attestationData[0] = AttestationStation.AttestationData({
+            about: bob,
+            key: bytes32("optimist.can-mint"),
+            val: bytes("true")
+        });
+        vm.prank(alice_admin);
+        attestationStation.attest(attestationData);
+
+        // assert bob is whitelisted
+        assertEq(optimist.isWhitelisted(bob), true);
+        // assert sally is not whitelisted
+        assertEq(optimist.isWhitelisted(sally), false);
+    }
+
+    /**
+     * @dev Should return the token id of the owner
+     */
+    function test_optimist_token_id_of_owner() external {
+        // whitelist bob
+        AttestationStation.AttestationData[]
+            memory attestationData = new AttestationStation.AttestationData[](1);
+        // we are using true but it can be any non null value
+        attestationData[0] = AttestationStation.AttestationData({
+            about: bob,
+            key: bytes32("optimist.can-mint"),
+            val: bytes("true")
+        });
+        vm.prank(alice_admin);
+        attestationStation.attest(attestationData);
+
+        // mint as bob
+        vm.prank(bob);
+        optimist.mint(bob);
+
+        // assert tokenid is correct
+        uint256 expectedId = 256;
+        assertEq(optimist.tokenIdOfAddress(address(bob)), expectedId);
+    }
+
+    /**
+     * @dev It should revert if anybody attemps token transfer
+     */
+    function test_optimist_sbt_transfer() external {
+        // whitelist bob
+        AttestationStation.AttestationData[]
+            memory attestationData = new AttestationStation.AttestationData[](1);
+        // we are using true but it can be any non null value
+        attestationData[0] = AttestationStation.AttestationData({
+            about: bob,
+            key: bytes32("optimist.can-mint"),
+            val: bytes("true")
+        });
+        vm.prank(alice_admin);
+        attestationStation.attest(attestationData);
+
+        // mint as bob
+        vm.prank(bob);
+        optimist.mint(bob);
+
```

Comment:
Is there a `transfer(address,uint256)` as well or is it just `transferFrom`?

neat trick:

```bash
forge inspect Optimist abi
forge inspect Optimist functionSelectors
```

Reply:
That's so dope 
<img width="760" alt="image" src="https://user-images.githubusercontent.com/35039927/207453264-92d5a128-2e5e-4073-af0b-7ccd91899687.png">